homeassistant/components/wemo/entity.py

```diff
@@ -169,27 +174,29 @@ async def async_added_to_hass(self) -> None:
         """Wemo device added to Home Assistant."""
         await super().async_added_to_hass()
 
-        registry = self.hass.data[WEMO_DOMAIN]["registry"]
-        await self.hass.async_add_executor_job(registry.register, self.wemo)
-        registry.on(self.wemo, None, self._subscription_callback)
+        @callback
+        def event_filter(event):
+            return self._device_id == event.data.get(CONF_DEVICE_ID)
+
+        self._stop_event_listener = self.hass.bus.async_listen(
```

Comment:
For internal communication in an integration we want to use our dispatch helper and not the event bus. The event bus is used to fire user facing events, that may be leveraged by device triggers.

Reply:
Done. The code now uses the dispatch helper for push notifications that impact the entities and uses events for user-facing events.